packages/eslint-plugin/CHANGELOG.md

```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### Enhancements
```

Comment:
As usual with ESLint rules, it's hard to tell if it's a breaking change, a new feature, or an enhancement 🤷‍♂️

Reply:
Enhanced breaking feature 💪

Reply:
Jokes aside… maybe a breaking change is best so folks aren't surprised by new lint problems?

Reply:
I plan to have a look at other plugins like the one from React or Jest.

Reply:
Jest:
- https://github.com/jest-community/eslint-plugin-jest/compare/v23.8.2...v23.9.0
- https://github.com/jest-community/eslint-plugin-jest/compare/v23.10.0...v23.11.0
- https://github.com/jest-community/eslint-plugin-jest/compare/v23.15.0...v23.16.0

When they add a new rule they put it under `Feature` section and it doesn't get enabled.

Reply:
React:
- https://github.com/yannickcr/eslint-plugin-react/releases/tag/v7.15.0
- https://github.com/yannickcr/eslint-plugin-react/releases/tag/v7.14.0

When new rule is added then it goes to `Added` section and doesn't get enabled in any config.

Reply:
Nice links, it looks like the last time they added rules to the recommended set it was flagged breaking and part of a major version change:

https://github.com/yannickcr/eslint-plugin-react/commit/c6c5e3b03c95c62a000b429d53d59a1e681f0f20

I'm wondering if there are advantages to add the rule first in a minor release, then adding it to the recommended rules in a major release later. There might be a few projects that enable it on their own and could provide some testing.

Would you like to go that route? I'm happy to remove it from the recommended rules for now.

Reply:
I've pulled the addition to recommended config into its own PR that can be landed later: https://github.com/WordPress/gutenberg/pull/27327